commit 3d38aa480219fc5d384030b841ceacbad795a934 (refs/changes/86/1386/3)
Author: Josh Wilsdon <jwilsdon@joyent.com>
Date:   2017-02-01T00:51:13-08:00 (2 years, 8 months ago)
    
    AGENT-1057 log setup for agents should move to a -setup service instead of the agent's startup script

diff --git a/npm/postinstall.sh b/npm/postinstall.sh
index 42a8058..c760b59 100755
--- a/npm/postinstall.sh
+++ b/npm/postinstall.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 set -o xtrace
@@ -43,6 +43,8 @@ subfile () {
   OUT=$2
   sed -e "s#@@PREFIX@@#$PREFIX#g" \
       -e "s/@@VERSION@@/$VERSION/g" \
+      -e "s#@@ROOT@@#$ROOT#g" \
+      -e "s/@@ENABLED@@/true/g" \
       $IN > $OUT
 }
 
@@ -52,6 +54,11 @@ function import_smf_manifest()
     if [[ $(uname -s) == "SunOS" ]]; then
         svccfg import $SMF_DIR/$AGENT.xml
     fi
+
+    subfile "$ROOT/smf/manifests/${AGENT}-setup.xml.in" "$SMF_DIR/${AGENT}-setup.xml"
+    if [[ $(uname -s) == "SunOS" ]]; then
+        svccfg import $SMF_DIR/${AGENT}-setup.xml
+    fi
 }
 
 function instance_exists()
diff --git a/package.json b/package.json
index 9144913..f74385b 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "firewaller",
     "description": "SmartDataCenter Firewalling Agent",
-    "version": "1.3.2",
+    "version": "1.4.0",
     "author": "Joyent (joyent.com)",
     "private": true,
     "dependencies": {
diff --git a/sbin/firewaller b/sbin/firewaller-setup
similarity index 72%
rename from sbin/firewaller
rename to sbin/firewaller-setup
index 4c1faea..5e761a6 100755
--- a/sbin/firewaller
+++ b/sbin/firewaller-setup
@@ -6,20 +6,17 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 #
-# SMF Startup script for the firewaller agent.
+# Setup script for the firewaller agent.
 #
 
 set -o xtrace
 
 . /lib/svc/share/smf_include.sh
 
-TOP=$(unset CDPATH; cd $(dirname $0)/../; pwd)
-
-
 # Add firewaller log rotation
 /usr/sbin/logadm -w firewaller_logs -C 168 -S 1g -c -p 1h \
     -t '/var/log/sdc/upload/firewaller_$nodename_%FT%H:%M:%S.log' \
@@ -29,9 +26,4 @@ TOP=$(unset CDPATH; cd $(dirname $0)/../; pwd)
 /usr/sbin/logadm -r smf_logs
 /usr/sbin/logadm -w smf_logs -C 8 -c -s 1m '/var/svc/log/*.log'
 
-
-/usr/bin/ctrun -l child -o noorphan ${TOP}/node/bin/node \
-    --abort_on_uncaught_exception \
-    ${TOP}/main.js &
-
 exit $SMF_EXIT_OK
diff --git a/smf/manifests/firewaller-setup.xml.in b/smf/manifests/firewaller-setup.xml.in
new file mode 100644
index 0000000..1b58ac4
--- /dev/null
+++ b/smf/manifests/firewaller-setup.xml.in
@@ -0,0 +1,55 @@
+<?xml version="1.0"?>
+<!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
+<!--
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+-->
+
+<!--
+    Copyright (c) 2017, Joyent, Inc.
+-->
+
+<service_bundle type="manifest" name="firewaller-agent-setup">
+  <service name="smartdc/agent/firewaller-agent-setup" type="service" version="@@VERSION@@">
+
+    <create_default_instance enabled="@@ENABLED@@"/>
+    <single_instance/>
+
+    <dependency name="network" grouping="require_all" restart_on="error" type="service">
+      <service_fmri value="svc:/milestone/network:default"/>
+    </dependency>
+
+    <exec_method
+      type="method"
+      name="start"
+      exec="@@ROOT@@/smf/method/firewaller-agent-setup %m"
+      timeout_seconds="600">
+      <method_context>
+        <method_credential user="root" group="staff"/>
+        <method_environment>
+          <envvar name="PATH" value="/usr/bin:/usr/sbin"/>
+        </method_environment>
+      </method_context>
+    </exec_method>
+
+    <exec_method type="method" name="stop" exec=":true" timeout_seconds="60" />
+
+    <property_group name="startd" type="framework">
+      <propval name="duration" type="astring" value="transient" />
+      <propval name="ignore_error" type="astring" value="core,signal" />
+    </property_group>
+
+    <property_group name="application" type="application">
+    </property_group>
+
+    <stability value="Evolving"/>
+
+    <template>
+      <common_name>
+        <loctext xml:lang="C">SmartDataCentre Firewaller Agent Setup</loctext>
+      </common_name>
+    </template>
+
+  </service>
+</service_bundle>
diff --git a/smf/manifests/firewaller.xml.in b/smf/manifests/firewaller.xml.in
index 0346f76..92c7c79 100644
--- a/smf/manifests/firewaller.xml.in
+++ b/smf/manifests/firewaller.xml.in
@@ -7,7 +7,7 @@
 -->
 
 <!--
-    Copyright (c) 2014, Joyent, Inc.
+    Copyright (c) 2017, Joyent, Inc.
 -->
 
 <!--
@@ -27,12 +27,12 @@
 <exec_method
   type="method"
   name="start"
-  exec="@@PREFIX@@/lib/node_modules/firewaller/sbin/firewaller"
+  exec="/usr/bin/ctrun -l child -o noorphan @@ROOT@@/node/bin/node --abort_on_uncaught_exception @@ROOT@@/main.js &amp;"
   timeout_seconds="60">
   <method_context>
     <method_credential user="root" group="staff"/>
     <method_environment>
-      <envvar name="PATH" value="@@PREFIX@@/bin:/usr/local/bin:/opt/local/bin:/usr/bin:/usr/sbin:/bin"/>
+      <envvar name="PATH" value="@@PREFIX@@/bin:/usr/bin:/usr/sbin"/>
     </method_environment>
   </method_context>
 </exec_method>
